client/src/components/uploadStoryForm/index.js

```diff
@@ -0,0 +1,115 @@
+/* eslint-disable */
+import React, { useState } from 'react';
+import { Modal, Button, Input } from 'antd';
+import './styles.css';
+import httpService from '../../services/httpService';
+// import axios from 'axios';
+
+const { TextArea } = Input;
+
+function StoryForm() {
+  const [fileInputState, setFileInputState] = useState('');
+  const [selectedFile, setSelectedFile] = useState();
+  const [previewSource, setPreviewSource] = useState('');
+  const [visible, setVisible] = useState(false);
+  const [story, setStory] = useState({
+    title: null,
+    content: null,
+    data: null,
+  });
+
+  const previewFile = (file) => {
+    const reader = new FileReader();
+    reader.readAsDataURL(file);
+    reader.onloadend = () => {
+      setPreviewSource(reader.result);
+    };
+  };
+
+  const handleFileInputChange = (e) => {
+    const file = e.target.files[0];
+    previewFile(file);
+    setSelectedFile(file);
+    setFileInputState(e.target.value);
+  };
+
+  const uploadImage = async (base64EncodedImage) => {
+    try {
+      await httpService.post('/api/v1/story', {
+        data: base64EncodedImage,
+          title: story.title,
+          content: story.content,
+      },
+       { headers: { 'Content-Type': 'application/json' }}
+      );
+      setFileInputState('');
+      setPreviewSource('');
+    } catch (err) {
+      throw new Error(`Error ${err}`);
+    }
+  };
+  const handleSubmitFile = (e) => {
+    e.preventDefault();
+    if (!selectedFile) return;
+    const reader = new FileReader();
+    reader.readAsDataURL(selectedFile);
+    setVisible(false);
+    reader.onloadend = () => {
+      uploadImage(reader.result);
+    };
+  };
+
+  return (
+    <>
+      <Button type="primary" onClick={() => setVisible(true)}>
+        أضف قصتك
+      </Button>
```

Comment:
this button will be out of this component, and visible and setVisisble will pass as a props to the component